### PR TITLE
Rebuild Subgraph Development Script

### PR DESCRIPTION
**Pull request by MiltonTulli**
Previously we were just waiting x amount of seconds expecting other services were going to be up but wasn't a great solution. 

In this pr I did a refactor of `dxdao-subgraph/scripts/dev.sh` to do the following: 
- Check if docker is open. If not open it. 
- Wait for docker to be ready
- Wait for hardhat node to be ready (we need that to run docker-compose)
- Update subgraph bytecodes and catch error nicely if doesnt' found local.json file
- Run docker compose up
- Wait for `graph-node` container to be running
- Run local graph deployment
- Once local build is ready, open graphQL playground in browser

### CODE REVIEW

File: apps/dxdao-subgraph/scripts/dev.sh [hunk 1]
```diff
@@ -1,10 +1,109 @@
...
 node ./scripts/updateBytecodes.js
-echo "Subgraph::: Waiting 30 seconds to dev-script to run"
-sleep 30
-echo "Subgraph::: Starting docker"
-docker compose up &
-echo "Subgraph::: waiting 60 seconds for docker"
-sleep 60
+
+echo "Subgraph::: Starting docker compose"
+docker-compose up &
```
Comment (dcrescimbeni) [inline apps/dxdao-subgraph/scripts/dev.sh:90]: In Linux, we need to execute `docker compose up` (without the dash).

With that change, everything works great :+1:

Comment (MiltonTulli) [reply apps/dxdao-subgraph/scripts/dev.sh]: Thanks for testing @dcrescimbeni ! I changed this. Can you test again if works for you in linux??

Comment (dcrescimbeni) [reply apps/dxdao-subgraph/scripts/dev.sh]: Doesn't seem to work after the refactor, but works after just the change to `docker compose up`

This is what I get:
```
$ yarn run updateBytecodes && yarn create-local && yarn deploy-local
$ node ./scripts/updateBytecodes.js
$ graph create mprasanjith/dxdao --node http://127.0.0.1:8020
✖ HTTP error creating the subgraph: ECONNREFUSED
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Review by rossneilson — APPROVED** [review]
Works for me

**Review by rossneilson — APPROVED** [review]

**Pull request merged**